keras_nlp/samplers/beam_sampler.py

```diff
@@ -77,7 +79,13 @@ def __call__(
         index=0,
         mask=None,
         end_token_id=None,
+        return_all_beams=None,
     ):
+        if return_all_beams is None:
```

Comment:
If we remove the call time argument, we can remove this whole if/else block too

Reply:
I tried that initially, but then as discussed above, it affects the unit test. Can definitely remove this if we are able to figure out the parameter passing in the test file.